expression/builtin_time.go

```diff
@@ -2505,18 +2505,59 @@ type timeFormatFunctionClass struct {
 }
 
 func (c *timeFormatFunctionClass) getFunction(args []Expression, ctx context.Context) (builtinFunc, error) {
-	sig := &builtinTimeFormatSig{newBaseBuiltinFunc(args, ctx)}
-	return sig.setSelf(sig), errors.Trace(c.verifyArgs(args))
+	if err := c.verifyArgs(args); err != nil {
+		return nil, errors.Trace(err)
+	}
+	bf, err := newBaseBuiltinFuncWithTp(args, ctx, tpDuration, tpString, tpString)
```

Comment:
OK. you should call `args[0].evalDuration()` and `args[1].evalString()` in `func (b *builtinTimeFormatSig) evalString(row []types.Datum) (string, bool, error)`